src/main/java/io/temporal/samples/hello/HelloSideEffect.java

```diff
@@ -0,0 +1,266 @@
+        throw new IllegalStateException("Random IDs not equal");
+      }
+
+      if (!sideEffectsRandomID.equals(savedSideEffectsRandomID)) {
+        throw new IllegalStateException("Sideffect random IDs not equal");
+      }
+
+      // This check will fail as unsafeRandomInt it was created not safely
+      // and breaks workflow determinism
+      if (unsafeRandomInt != savedUnsafeRandomInt) {
+        return "Unsafely created random numbers are not equal";
+      }
+
+      return result;
+    }
+  }
+
+  /** Simple activity implementation. */
+  static class SideEffectActivitiesImpl implements SideEffectActivities {
+    @Override
+    public String doSomething(String input) {
+      return "Received input: " + input;
+    }
+  }
+
+  /**
+   * With our Workflow and Activities defined, we can now start execution. The main method starts
+   * the worker and then the workflow.
+   */
+  public static void main(String[] args) {
+
+    // Define the workflow service.
+    WorkflowServiceStubs service = WorkflowServiceStubs.newInstance();
+
+    /*
+     * Define the workflow client. It is a Temporal service client used to start, signal, and query
+     * workflows
+     */
+    WorkflowClient client = WorkflowClient.newInstance(service);
+
+    /*
+     * Define the workflow factory. It is used to create workflow workers for a specific task queue.
+     * For this example we set the setWorkflowHostLocalTaskQueueScheduleToStartTimeout to zero
+     * (to disable sticky execution)
+     */
+    WorkerFactory factory =
+        WorkerFactory.newInstance(
+            client,
+            WorkerFactoryOptions.newBuilder()
+                .setWorkflowHostLocalTaskQueueScheduleToStartTimeout(Duration.ZERO)
+                .build());
+
+    /*
+     * Define the workflow worker. Workflow workers listen to a defined task queue and process
+     * workflows and activities.
+     */
+    Worker worker = factory.newWorker(TASK_QUEUE);
+
+    /*
+     * Register our workflow implementation with the worker.
+     * Workflow implementations must be known to the worker at runtime in
+     * order to dispatch workflow tasks.
+     */
+    worker.registerWorkflowImplementationTypes(SideEffectWorkflowImpl.class);
+
+    /*
+     Register our workflow activity implementation with the worker. Since workflow activities are
+     stateless and thread-safe, we need to register a shared instance.
+    */
+    worker.registerActivitiesImplementations(new SideEffectActivitiesImpl());
+
+    /*
+     * Start all the workers registered for a specific task queue.
+     * The started workers then start polling for workflows and activities.
+     */
+    factory.start();
+
+    // Create the workflow client stub. It is used to start our workflow execution.
+    SideEffectWorkflow workflow =
+        client.newWorkflowStub(
+            SideEffectWorkflow.class,
+            WorkflowOptions.newBuilder()
+                .setWorkflowId(WORKFLOW_ID)
```

Comment:
Why do you set workflow id in this sample?

Reply:
when people run samples to learn i think its easier to see which workflow run is/was associated with which example.